### DISCUSSION TITLE
Home Assistant Integration: BMW Connected Drive works only partially (lock fails)

### DISCUSSION THREAD

**Discussion by nikoedii** [bimmerconnected/bimmer_connected#663] [Q&A]
Hi there, I can successfully use the BMW Connected Drive app with my 2015 F34 (limited scope: location, lock, horn, lights - but especially the first two come in handy). I managed to install the Home Assistant integration and can use it for most of the functions (location, horn, lights..) BUT NOT the lock/unlock. The lock icon is greyed out, and even when lock / unlock icons are presented (and clickable), nothing happens. Any idea what I am doing wrong? Logs attached.
THanks for your help and best regards
Niko
[config_entry-bmw_connected_drive-01JBCRA8NWN54GMQ4XA4ZWFQ4J.json](https://github.com/user-attachments/files/17561604/config_entry-bmw_connected_drive-01JBCRA8NWN54GMQ4XA4ZWFQ4J.json)

![image](https://github.com/user-attachments/assets/4789ad11-5802-4c2e-ae96-7b07be260ba9)
![image](https://github.com/user-attachments/assets/266f2187-8aae-4ef6-b833-3350b44820dc)

**Comment by rikroe** [comment]
That's normal for the first-gen ConnectedDrive cars. Pressing the lock/unlock button will send the command to the car (if you're standing next to it, it'll open after a minute or so).
However as we're not able to show the state of the lock (we don't get this info back), the lock state itself will always be unknown.
